src/lib/Libutil/pbs_aes_encrypt.c

```diff
@@ -90,27 +100,36 @@ pbs_encrypt_pwd(char *str, int *credtype, char **credbuf, size_t *credlen)
 int
 pbs_decrypt_data(char *crypted, int credtype, size_t len, char **uncrypted, size_t *outlen)
 {
-        EVP_CIPHER_CTX ctx;
         char *cblk;
         int plen, len2 = 0;
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#define CIPHER_CONTEXT_INIT(v) EVP_CIPHER_CTX_init(v)
+#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_cleanup(v)
+        EVP_CIPHER_CTX real_ctx;
+        EVP_CIPHER_CTX *ctx = &real_ctx;
+#else
+#define CIPHER_CONTEXT_INIT(v) v = EVP_CIPHER_CTX_new(); if (!v) return -1;
+#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_free(v)
+        EVP_CIPHER_CTX *ctx = NULL;
+#endif
```

Comment:
You shouldn't have to redefine CIPHER_CONTEXT_INIT and CIPHER_CONTEXT_CLEAN here because they were already defined above. I would expect the compiler to complain about their redefinition. Could you please check your build log?

Other than that, the updated changes look good to me.